tests/unit_tests/test_temp_interp.py

```diff
@@ -0,0 +1,155 @@
+from math import isnan
+import os
+from pathlib import Path
+
+from uncertainties import ufloat
+import numpy as np
+import openmc.data
+from openmc.data import K_BOLTZMANN
+from openmc.stats import Uniform
+import pytest
+
+
+def make_fake_cross_section():
+    """Create fake U235 nuclide
+
+    This nuclide is designed to have k_inf=1 at 300 K, k_inf=2 at 600 K, and
+    k_inf=1 at 900 K. The absorption cross section is also constant with
+    temperature so as to make the true k-effective go linear with temperature.
+    """
+
+    def isotropic_angle():
+        return openmc.data.AngleDistribution(
+            [E_min, E_max],
+            [Uniform(-1., 1.), Uniform(-1., 1.)]
+        )
+
+    def cross_section(value):
+        return openmc.data.Tabulated1D(
+            energy,
+            value*np.ones_like(energy)
+        )
+
+    temperatures = (300, 600, 900)
+
+    u235_fake = openmc.data.IncidentNeutron(
+        'U235', 92, 235, 0, 233.0248, [T*K_BOLTZMANN for T in temperatures]
+    )
+
+    # Create energy grids
+    E_min, E_max = 1e-5, 20.0e6
+    energy = np.logspace(np.log10(E_min), np.log10(E_max))
+    for T in temperatures:
+        u235_fake.energy[f'{T}K'] = energy
```

Comment:
Python 3.5 reached end of life in September of last year. Many other third-party packages have already dropped support, e.g. [numpy](https://numpy.org/devdocs/release/1.19.0-notes.html) (3.6+), [scipy](https://github.com/scipy/scipy/blob/master/INSTALL.rst.txt#prerequisites) (3.7+), and [pandas](https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#python-version-support) (3.7+). I'd propose that we bump the minimum Python version to 3.6 so that developers will be allowed to use f-strings and other goodness from 3.6, but perhaps it's better to do that as a separate PR so for now I'll revert to the `format` method.

Reply:
Yeah, agreed. We really should update to 3.6+ throughout.